server/server.go

```diff
@@ -0,0 +1,123 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"net/http"
+
+	"github.com/Clever/cron-admin/db"
+)
+
+func byteHandler(handler func(*http.Request) ([]byte, error)) func(http.ResponseWriter, *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		data, err := handler(req)
+		if err != nil {
+			fmt.Printf("Received handler err: %s\n", err)
+		}
+		rw.Write(data)
+	}
+}
+
+func jsonHandler(handler func(*http.Request) (interface{}, error)) func(http.ResponseWriter, *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		wrappedHandler := func(req *http.Request) ([]byte, error) {
+			resp, err := handler(req)
+			if err != nil {
+				return nil, err
+			}
+			data, err := json.Marshal(resp)
+			if err != nil {
+				return nil, err
+			}
+			rw.Header().Add("Content-Type", "application/json; charset=utf-8")
+			return data, nil
+		}
+		byteHandler(wrappedHandler)(rw, req)
+	}
+}
+
+func SetupHandlers(r *mux.Router, db *mongodb.DB) {
+	r.HandleFunc("/active-jobs", jsonHandler(func(req *http.Request) (interface{}, error) {
+		defer req.Body.Close()
+		activeJobs, err := db.GetDistinctActiveJobs()
+		if err != nil {
+			fmt.Println(err)
+			return nil, err
+		}
+		return activeJobs, nil
+	})).Methods("GET")
+
+	r.HandleFunc("/job-details", jsonHandler(func(req *http.Request) (interface{}, error) {
+		defer req.Body.Close()
+		job := req.URL.Query().Get("job")
+		jobDetails, err := db.GetJobDetails(job)
+		if err != nil {
+			fmt.Println(err)
+			return nil, err
+		}
+		return jobDetails, nil
+	})).Methods("GET")
+
+	r.HandleFunc("/modify-job/{job_id}", jsonHandler(func(req *http.Request) (interface{}, error) {
```

Comment:
`PATCH /jobs/{id}` with a JSON request body containing updates would be more RESTful